client/python/modeldb/basic/ModelDbSyncerBase.py

```diff
@@ -371,4 +373,46 @@ def get_dataset_for_tag(self, data_tag):
             data_tag = "default"
         return self.datasets[data_tag]
 
-    # TODO: do we want a sync all?
+    def dataset_from_dict(self, dataset_dict):
+        filename = dataset_dict[metadata_constants.DATASET_FILENAME_KEY]
+        metadata = dataset_dict[metadata_constants.DATASET_METADATA_KEY] \
+            if metadata_constants.DATASET_METADATA_KEY in dataset_dict else {}
```

Comment:
Can be done shortly: `dataset_dict.get(metadata_constants.DATASET_METADATA_KEY, {})`